addons/sale/tests/test_sale_report.py

```diff
@@ -17,66 +15,149 @@ class TestSaleReportCurrencyRate(SaleCommon):
-                'name': fields.Date.today(),
-                'company_rate': 3.0,
-                'currency_id': currency_eur.id,
-                'company_id': self.company.id,
-            })
-            eur_order = self.env['sale.order'].with_company(self.company).create({
-                'partner_id': self.partner.id,
-                'pricelist_id': eur_pricelist.id,
-                'date_order': fields.Date.today(),
-                'order_line': [
-                    Command.create({
-                        'product_id': self.product.id,
-                        'product_uom_qty': 1.0,
-                    }),
-                ]
-            })
-            # the sales order's amount is in foreign currency
-            self.assertEqual(eur_order.amount_total, 60.0)
-
-        # the report should show the amount in company currency
-        report_line = self.env['sale.report'].sudo().search(
-            [('order_id', '=', eur_order.id)])
-        self.assertEqual(report_line.price_total, 20.0)
-
-    def test_sale_report_company_currency(self):
+        # Activate a third currency and create corresponding pricelists.
+        currency_ars = self._enable_currency('ARS')
         usd_pricelist = self.env['product.pricelist'].create({
             'name': 'Pricelist (USD)',
-            'currency_id': self.company.currency_id.id,
+            'currency_id': self.usd_company.currency_id.id,
+        })
+        eur_pricelist = self.env['product.pricelist'].create({
+            'name': 'Pricelist (EUR)',
+            'currency_id': self.eur_company.currency_id.id,
+        })
+        ars_pricelist = self.env['product.pricelist'].create({
+            'name': 'Pricelist (ARS)',
+            'currency_id': currency_ars.id,
+        })
+
+        # Create rates.
+        self.env['res.currency.rate'].create({
+            'name': fields.Date.today(),
+            'rate': 0.5,
+            'currency_id': self.eur_company.currency_id.id,
+            'company_id': self.usd_company.id,
+        })
+        self.env['res.currency.rate'].create({
+            'name': fields.Date.today(),
+            'rate': 10.0,
+            'currency_id': currency_ars.id,
+            'company_id': self.usd_company.id,
         })
 
-        usd_order = self.env['sale.order'].with_company(self.company).create({
+        self.env['res.currency.rate'].create({
+            'name': fields.Date.today(),
+            'rate': 2,
+            'currency_id': self.usd_company.currency_id.id,
+            'company_id': self.eur_company.id,
+        })
+        self.env['res.currency.rate'].create({
+            'name': fields.Date.today(),
+            'rate': 20.0,
+            'currency_id': currency_ars.id,
+            'company_id': self.eur_company.id,
+        })
+
+        self.env['res.currency.rate'].create({
+            'name': fields.Date.today(),
+            'rate': 10.0,
+            'currency_id': currency_ars.id,
+            'company_id': 1,
```

Comment:
1? oO Which company is that? 
Could you instead indicates clearly the company ? It's quite unclear as is, and might at some point not be the one you expect it to be.